File: easy/141.Linked List Cycle/rio/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let firstPointer = head;
    let secondPointer = head;

    while(firstPointer && firstPointer.next){
        firstPointer = firstPointer.next.next;
        secondPointer = secondPointer.next;

        if(firstPointer === secondPointer) return true;
    }
    return false;  
};

File: easy/141.Linked List Cycle/rio/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 struct ListNode* newNode(int value){
    struct ListNode* node = (struct ListNode*)malloc(sizeof(struct ListNode));
    node->val = value;
    node->next = NULL;
    return node;
 }
bool hasCycle(struct ListNode *head) {
    struct ListNode *first = head;
    struct ListNode *second = head;

    while(first != NULL && first->next != NULL){
        first = first->next->next;
        second = second->next; 
    
        if(first == second){
            return true;
        }
    }
    return false;
}